pkg/pool-hooks/contracts/StableSurgeHookExample.sol

```diff
@@ -0,0 +1,235 @@
+    using FixedPoint for uint256;
+    // Only pools from a specific factory are able to register and use this hook.
+    address private immutable _allowedFactory;
+    // Defines the range in which surging will not occur; meaning surging will occur when:
+    // B(i-after swap) / (sum(B(n-after swap))) > 1/n + gamma
+    uint256 private _threshold;
+    // An amplification coefficient to amplify the degree a fee increases upon the threshold being triggered
+    uint256 private _surgeCoefficient;
+
+    /**
+     * @notice A new `StableSurgeHookExample` contract has been registered successfully.
+     * @dev If the registration fails the call will revert, so there will be no event.
+     * @param hooksContract This contract
+     * @param factory The factory (must be the allowed factory, or the call will revert)
+     * @param pool The pool on which the hook was registered
+     */
+    event StableSurgeHookExampleRegistered(
+        address indexed hooksContract,
+        address indexed factory,
+        address indexed pool
+    );
+
+    /**
+     * @notice The threshold has been changed in a `StableSurgeHookExample` contract.
+     * @dev Note, the initial threshold is set on deployment and an event is emitted.
+     * @param hooksContract This contract
+     * @param threshold The new threshold
+     */
+    event ThresholdChanged(address indexed hooksContract, uint256 indexed threshold);
+
+    /**
+     * @notice The surgeCoefficient has been changed in a `StableSurgeHookExample` contract.
+     * @dev Note, the initial surgeCoefficient is set on deployment and an event is emitted.
+     * @param hooksContract This contract
+     * @param surgeCoefficient The new surgeCoefficient
+     */
+    event SurgeCoefficientChanged(address indexed hooksContract, uint256 indexed surgeCoefficient);
+
+    constructor(
+        IVault vault,
+        address allowedFactory,
+        uint256 threshold,
+        uint256 surgeCoefficient
+    ) VaultGuard(vault) Ownable(msg.sender) {
+        _allowedFactory = allowedFactory;
+        _threshold = threshold;
+        _surgeCoefficient = surgeCoefficient;
+        emit ThresholdChanged(address(this), threshold);
+        emit SurgeCoefficientChanged(address(this), surgeCoefficient);
+    }
+
+    /// @inheritdoc IHooks
+    function getHookFlags() public pure override returns (HookFlags memory hookFlags) {
+        hookFlags.shouldCallComputeDynamicSwapFee = true;
+    }
+
+    /// @inheritdoc IHooks
+    function onRegister(
+        address factory,
+        address pool,
+        TokenConfig[] memory,
+        LiquidityManagement calldata
+    ) public override onlyVault returns (bool) {
+        // This hook implements a restrictive approach, where we check if the factory is an allowed factory and if
+        // the pool was created by the allowed factory.
+        emit StableSurgeHookExampleRegistered(address(this), factory, pool);
```

Comment:
Is this best practice I should add to scaffold examples? 

Looks like the `PoolRegistered` event already contains all the same info, so maybe this just enables easier indexing / filtering?

https://github.com/balancer/balancer-v3-monorepo/blob/184494da6351fc9daf61b738ea2ed7da63694ee8/pkg/vault/contracts/VaultExtension.sol#L316-L326

Reply:
I introduced this; there are already PRs in docs about this, and the monorepo has these added. It's up to the subgraph / front end people if they find this useful, but I thought this was a simpler, focused event in case we have very popular hooks that are used a lot. Since there could be multiple instances of the same hook with different parameters (in some cases).

Reply:
Would move the event down after the check, though. No sense emitting and then reverting, plus it separates the comment from the logic.

Reply:
Based of Daniel conclusion during the Office Hours chat it seems like we should leave this as is?

Reply:
Sure; it's fine. Clearer this way, and we don't need to optimize for a case when it reverts.